src/test/ui/issues/issue-11692-2.stderr

```diff
@@ -3,6 +3,8 @@ error: cannot find macro `test` in this scope
    |
 LL |     concat!(test!());
    |             ^^^^
+   |
+   = note: `test` is in scope, but it is an attribute
```

Comment:
Seeing this case makes me wonder, would some users get confused with what an attribute is? Things like functions and structs should be fine, but I'm wondering if we should show "how to use them". I also wonder whether we should be showing their `def_span` (which we can easily do if we also look beyond the macros to get the item's def id).

Reply:
We could also just not show the note in some cases if it wasn't explicitly imported. Mixing up derive and attribute macros should probably always show the note, but if a regular macro is involved, it might be good to only show the note if there's a `use std::test;` or something to point at.

Reply:
It now shows:

```
   = note: `test` is in scope, but it is an attribute: `#[test]`
```
and
```
   = note: `MyTrait` is in scope, but it is a derive macro: `#[derive(MyTrait)]`
```